opesci/grid.py

```diff
@@ -20,6 +20,9 @@ class Grid:
     # property field in the derived grid class
     template_keys = []
 
+    # Default compiler is GNU
+    compiler = GNUCompiler()
```

Comment:
The default value is set because `grid.generate()` might access the `compiler` property for platform-specific pragmas like `ivdep` before `grid.compile()` is invoked. We might actually want a setter method for translating string to `Compiler` object, rather than hijack `grid.compile()`?